### PR TITLE
docs: add TypeDocs auto generated API Reference

### LINKED ISSUE #132: Investigate TypeDocs for API documentation
Investigate using https://typedoc.org/example/index.html for documenting the API

Currently, we are missing a large amount of API docs, and considering that the API is relatively complex we should try fix this.

### PR DESCRIPTION
**Pull request by oliverbutler**
fixes #132

### CODE REVIEW

File: libs/ts-rest/solid-query/package.json [hunk 1]
```diff
@@ -34,5 +34,9 @@
   "main": "index.cjs",
   "module": "index.js",
   "types": "./src/index.d.ts",
-  "type": "module"
+  "type": "module",
```
Comment (michaelangrivera) [inline libs/ts-rest/solid-query/package.json:37]: Will `module` affect any of the rollup compilation? Want to make sure there are no regressions.

**Comment by michaelangrivera** [comment]
LGTM, however, can you update or create a README somewhere on _how_ to update these API docs? So future maintainers can reference it to auto-update the docs when they make changes? IE, any npm commands to run to update it, etc.

File: libs/ts-rest/solid-query/package.json (hunk 1 above)
Comment (oliverbutler) [reply libs/ts-rest/solid-query/package.json:37]: This isn't new, I believe we had many tanstack lib issues without it

File: apps/docs/docs/api/modules/ts_rest_nest.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+#### Type parameters
+
+| Name | Type |
+| :------ | :------ |
+| `TRoute` | extends `AppRoute` |
+
+#### Defined in
+
+[libs/ts-rest/nest/src/lib/api.decorator.ts:37](https://github.com/oliverbutler/tscont/blob/a69280b/libs/ts-rest/nest/src/lib/api.decorator.ts#L37)
```
Comment (michaelangrivera) [inline apps/docs/docs/api/modules/ts_rest_nest.md:27]: maybe this should point to `https://github.com/ts-rest` somehow?

File: libs/ts-rest/solid-query/package.json (hunk 1 above)
Comment (michaelangrivera) [reply libs/ts-rest/solid-query/package.json:37]: my b, confused the github diff

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -32,3 +32,13 @@ NX will run all of the tests for the affected apps, helping you save some time
...
+
+## Documentation
+
+The documentation is built using [Docusaurus](https://docusaurus.io/). You can find the source code in the `docs` directory.
+
+```
+pnpm nx run docs:serve
+```
+
+Note, when you run this command the TypeDocs (in the `docs/api` directory) will be generated automatically!
```
Comment (michaelangrivera) [inline CONTRIBUTING.md:44]: 🥳

File: apps/docs/docs/api/modules/ts_rest_nest.md (hunk 2 above)
Comment (oliverbutler) [reply apps/docs/docs/api/modules/ts_rest_nest.md:27]: Yeah, you're right. It's annoying, every time I make a change in docs it outputs this link.

Which by itself is weird, its the OG path to ts-rest, and that name isn't _anywhere_ in the repo. So I'm not sure why it's doing that.

**Review by michaelangeloio — APPROVED** [review]

**Pull request merged**